bacon/controller/AddTransactionViewController.swift

```diff
@@ -92,21 +127,64 @@ class AddTransactionViewController: UIViewController {
         return amountDecimal ?? Constants.defaultAmount
     }
 
+    private func captureDescription() -> String {
+        let userInput = descriptionField.text
+        return userInput ?? Constants.defaultDescription
+    }
+
+    private func capturePhoto() -> CodableUIImage? {
+        guard let image = photo else {
+            return nil
+        }
+        return CodableUIImage(image)
+    }
+
+    private func captureLocation() {
+        guard let location = locationManager.location else {
+            return
+        }
+        // Display
+        geoCoder.reverseGeocodeLocation(location) { placemarks, _ in
+            if let place = placemarks?.first {
+                self.locationLabel.text = "\(place)" // need to format a bit
+            }
+        }
+        userLocation = CodableCLLocation(location)
+    }
+
     private func setExpenditureType() {
         transactionType = .expenditure
-        typeLabel.text = "-"
+        typeLabel.text = "- $"
         typeLabel.textColor = UIColor.red
         categoryLabel.textColor = UIColor.red
     }
 
     private func setIncomeType() {
         transactionType = .income
-        typeLabel.text = "+"
+        typeLabel.text = "+ $"
```

Comment:
Bell and whistle suggestion: localization to different currency symbols, e.g. if UK then use the pound symbol